docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+  Uses Geneve overlay.
+* **Localnet Type Network** - An OVN-Kubernetes topology rendered into OVN where pods connect to a per-node logical switch
+  that is directly wired to the underlay.
+
+## Goals
+
+* Provide a configurable way to indicate that a pod should be connected to a user-defined network of a specific type as a
+  primary interface.
+* The primary network may be configured as a layer 3 or layer 2 type network.
+* Allow networks to have overlapping pod IP address space. This range may not overlap with the default cluster subnet
+  used for allocating pod IPs on the cluster default network today.
+* The cluster default primary network defined today will remain in place as the default network pods attach to. The cluster
+  default network will continue to serve as the primary network for pods in a namespace that has no primary user-defined network. Pods
+  with primary user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+  Pods with primary user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+  connected to the user-defined network. Pods with primary user-defined networks will use the user-defined network as their default
+  gateway.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features on user-defined primary networks where possible:
+    * EgressQoS
+    * EgressService
+    * EgressIP
+    * Load Balancer and NodePort Services, as well as services with External IPs.
+* In addition to ingress service support, there will be support for Kubernetes services in user-defined networks. The
+  scope of reachability to that service as well as endpoints selected for that service will be confined to the network
+  and corresponding namespace(s) where that service was created.
+* Support for pods to continue to have access to the cluster default primary network for DNS and KAPI service access.
+* Kubelet healthchecks/probes will still work on all pods.
+
+## Non-Goals
+
+* Allowing different service CIDRs to be used in different networks.
+* Localnet will not be supported initially for primary networks.
+* Allowing multiple primary networks per namespace.
+* Hybrid overlay support on user-defined networks.
+
+## Future-Goals
+
+* DNS lookup for pods returning records for IPs on the user-defined network. In the first phase DNS will return the pod
+  IP on the cluster default network instead.
+* Admin ability to configure networks to have access to all services and/or expose services to be accessible from all
+  networks.
+* Ability to advertise user-defined networks to external networks using BGP/EVPN. This will enable things like:
+    * External -> Pod ingress per VRF (Ingress directly to pod IP)
+    * Multiple External Gateway (MEG) in a BGP context, with ECMP routes
```

Comment:
can we include Egress Services and Host-Networked Pods here? as future goals.
so far we have not implemented both of these though I know your enhancement meant to design it for both